segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/DefragmentOperation.java

```diff
@@ -144,11 +147,28 @@
                             CompletableFuture<Void> f;
                             if (chunksToConcat.size() > 1) {
                                 // Concat
-                                f = concatChunks();
+                                f = concatChunks()
+                                .handleAsync((vv, ex) -> {
+                                    if (null != ex) {
+                                        ex = Exceptions.unwrap(ex);
+                                        if (ex instanceof InvalidOffsetException) {
+                                            // Skip ahead by 1 chunk.
+                                            targetChunkName = chunksToConcat.get(1).getName();
+                                            chunksToConcat.clear();
+                                            skipFailed.set(true);
+                                            return null;
+                                        }
+                                        throw new CompletionException(ex);
```

Comment:
What other types of exceptions are expected here and why they are bubbled up (conversely to `InvalidOffsetException`)?

Reply:
We are specifically handling only the `InvalidOffsetException` and nothing else. 
For others we just bubble them up.